src/app/directives/restrict-special-char.directive.ts

```diff
@@ -0,0 +1,25 @@
+import { Directive, HostListener, Input } from '@angular/core';
+
+@Directive({
+  selector: '[appRestrictSpecialChars]'
+})
+export class RestrictSpecialCharDirective {
+  @Input() restrictedCharCodes: any[];
+
+  constructor() { }
+
+  @HostListener('keypress') onkeypress(e) {
+    const event = e || window.event;
+    if (event) {
+      return this.isRestrictedChar(event);
+    }
+  }
+
+  isRestrictedChar(event) {
+    if (this.restrictedCharCodes?.length === 0) {
+      this.restrictedCharCodes = ['34'];
```

Comment:
Why `34` as string in the array? It is failing in this.restrictedCharCodes?.includes(charCode); because event.keyCode is a number. 


you can make it more simple. like
```
restrictedCharCodes = [34];  // 34 = double quote(") key code

  constructor() { }

  @HostListener('keypress') onkeypress(e) {
    const event = e || window.event;
    if (event) {
      return this.isRestrictedChar(event);
    }
  }

  isRestrictedChar(event) {
    const charCode = (event.which) ? event.which : event.keyCode;
    return !this.restrictedCharCodes?.includes(charCode);
  }
```

Reply:
> Why 34 as string in the array?

Right, it should be a number. Actually, that was mistakenly written as string in the previous commit. It was a number before. I'll fix it in the next commit.

> It is failing in this.restrictedCharCodes?.includes(charCode); because event.keyCode is a number.

No, It was failing because `restrictedCharCodes` has undefined value (it is not entering in the `if` code block). The reason behind this is that- for now we are not passing any value to `restrictedCharCodes` from any parent component.
I'll add a check for this.

After reading my above answers, it will be clear to you that I can't change the code as above (in your comment).